sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -172,13 +172,17 @@ trait CheckAnalysis extends PredicateHelper {
             failAnalysis("Null-aware predicate sub-queries cannot be used in nested " +
               s"conditions: $condition")
 
+          case f @ Filter(condition, _) if PlanHelper.specialExpressionInUnsupportedOperator(f) =>
```

Comment:
do we still need this?

Reply:
@cloud-fan Actually we don't need it. However, since its a user facing message, i feel, its better if we issue targeted error message. I was thinking, we we should add more individual cases as we learn about them here while still having a generic catcher. 
If you think otherwise, then i will remove it. Let me know, please.